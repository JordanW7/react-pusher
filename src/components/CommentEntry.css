.commententry-user {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: left;
  align-content: center;
}

.commententry-avatar {
	margin: 5px;
}
