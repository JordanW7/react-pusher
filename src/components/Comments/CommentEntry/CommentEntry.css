.commententry-user {
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  justify-content: left;
  align-content: center;
}

.commententry-avatar {
  margin: 5px;
}

.commententry {
  border-bottom: 1px dashed rgba(255, 255, 255, 0.2);
}
