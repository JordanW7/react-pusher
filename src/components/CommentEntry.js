import React from 'react';
import { Row, Col, Avatar } from "antd";
import './CommentEntry.css';

const CommentEntry = () => {
	return (
		<Row>
			<div>
			<Col span={24} className="commententry-user">
				<Avatar className="commententry-avatar">A</Avatar>
				Anthony Smith says:
			</Col>
			</div>
			<Col span={24}>
				<p>This is my Comment!</p>
			</Col>
		</Row>
		);
}

export default CommentEntry;
