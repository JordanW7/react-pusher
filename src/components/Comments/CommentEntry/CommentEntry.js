import React from "react";
import { Row, Col, Avatar } from "antd";
import moment from "moment";
import "./CommentEntry.css";

const CommentEntry = ({ name, comment, img, time }) => {
  const avatarimg = (
    <img src={img} className="commententry-avatar" alt={name} />
  );
  const avatar =
    name === "You " ? (
      <Avatar
        style={{ color: "#ffffff", backgroundColor: "#fb3f6c" }}
        className="commententry-avatar"
      >
        {name}
      </Avatar>
    ) : (
      <Avatar
        style={{ color: "#ffffff", backgroundColor: "#00bfff" }}
        className="commententry-avatar"
      >
        {name}
      </Avatar>
    );
  return (
    <Row className="commententry">
      <div className="commententry-user">
        <Col span={18}>
          {img ? avatarimg : avatar}
          {name}:
        </Col>
        <Col span={6} className="commententry-user">
          {`(${moment(time).fromNow()})`}
        </Col>
      </div>
      <Col span={24}>
        <p>{comment}</p>
      </Col>
    </Row>
  );
};

export default CommentEntry;
