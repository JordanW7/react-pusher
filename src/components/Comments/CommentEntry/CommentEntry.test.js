import React from "react";
import { shallow } from "enzyme";
import CommentEntry from "./CommentEntry";

it("expect to render Comment Entry component", () => {
  expect(
    shallow(
      <CommentEntry key="1" name="You" img="" comment="Hello" time="123" />
    )
  ).toMatchSnapshot();
});
