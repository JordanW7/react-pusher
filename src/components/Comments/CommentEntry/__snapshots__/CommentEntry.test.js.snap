// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Comment Entry component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentEntry
    comment="Hello"
    img=""
    name="You"
    time="123"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="commententry-user"
        >
          <Col
            span={18}
          >
            <Avatar
              className="commententry-avatar"
              prefixCls="ant-avatar"
              shape="circle"
              size="default"
              style={
                Object {
                  "backgroundColor": "#00bfff",
                  "color": "#ffffff",
                }
              }
            >
              You
            </Avatar>
            You
            :
          </Col>
          <Col
            className="commententry-user"
            span={6}
          >
            (1896 years ago)
          </Col>
        </div>,
        <Col
          span={24}
        >
          <p>
            Hello
          </p>
        </Col>,
      ],
      "className": "commententry",
      "gutter": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Col
              span={18}
            >
              <Avatar
                className="commententry-avatar"
                prefixCls="ant-avatar"
                shape="circle"
                size="default"
                style={
                  Object {
                    "backgroundColor": "#00bfff",
                    "color": "#ffffff",
                  }
                }
              >
                You
              </Avatar>
              You
              :
            </Col>,
            <Col
              className="commententry-user"
              span={6}
            >
              (1896 years ago)
            </Col>,
          ],
          "className": "commententry-user",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Avatar
                  className="commententry-avatar"
                  prefixCls="ant-avatar"
                  shape="circle"
                  size="default"
                  style={
                    Object {
                      "backgroundColor": "#00bfff",
                      "color": "#ffffff",
                    }
                  }
                >
                  You
                </Avatar>,
                "You",
                ":",
              ],
              "span": 18,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "You",
                  "className": "commententry-avatar",
                  "prefixCls": "ant-avatar",
                  "shape": "circle",
                  "size": "default",
                  "style": Object {
                    "backgroundColor": "#00bfff",
                    "color": "#ffffff",
                  },
                },
                "ref": null,
                "rendered": "You",
                "type": [Function],
              },
              "You",
              ":",
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "(1896 years ago)",
              "className": "commententry-user",
              "span": 6,
            },
            "ref": null,
            "rendered": "(1896 years ago)",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <p>
            Hello
          </p>,
          "span": 24,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Hello",
          },
          "ref": null,
          "rendered": "Hello",
          "type": "p",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="commententry-user"
          >
            <Col
              span={18}
            >
              <Avatar
                className="commententry-avatar"
                prefixCls="ant-avatar"
                shape="circle"
                size="default"
                style={
                  Object {
                    "backgroundColor": "#00bfff",
                    "color": "#ffffff",
                  }
                }
              >
                You
              </Avatar>
              You
              :
            </Col>
            <Col
              className="commententry-user"
              span={6}
            >
              (1896 years ago)
            </Col>
          </div>,
          <Col
            span={24}
          >
            <p>
              Hello
            </p>
          </Col>,
        ],
        "className": "commententry",
        "gutter": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Col
                span={18}
              >
                <Avatar
                  className="commententry-avatar"
                  prefixCls="ant-avatar"
                  shape="circle"
                  size="default"
                  style={
                    Object {
                      "backgroundColor": "#00bfff",
                      "color": "#ffffff",
                    }
                  }
                >
                  You
                </Avatar>
                You
                :
              </Col>,
              <Col
                className="commententry-user"
                span={6}
              >
                (1896 years ago)
              </Col>,
            ],
            "className": "commententry-user",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Avatar
                    className="commententry-avatar"
                    prefixCls="ant-avatar"
                    shape="circle"
                    size="default"
                    style={
                      Object {
                        "backgroundColor": "#00bfff",
                        "color": "#ffffff",
                      }
                    }
                  >
                    You
                  </Avatar>,
                  "You",
                  ":",
                ],
                "span": 18,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "You",
                    "className": "commententry-avatar",
                    "prefixCls": "ant-avatar",
                    "shape": "circle",
                    "size": "default",
                    "style": Object {
                      "backgroundColor": "#00bfff",
                      "color": "#ffffff",
                    },
                  },
                  "ref": null,
                  "rendered": "You",
                  "type": [Function],
                },
                "You",
                ":",
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "(1896 years ago)",
                "className": "commententry-user",
                "span": 6,
              },
              "ref": null,
              "rendered": "(1896 years ago)",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <p>
              Hello
            </p>,
            "span": 24,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hello",
            },
            "ref": null,
            "rendered": "Hello",
            "type": "p",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
